src/blob/index/bptree/core.rs

```diff
@@ -148,8 +152,8 @@ impl FileIndexTrait for BPTreeFileIndex {
     }
 }
 
-impl BPTreeFileIndex {
-    async fn find_leaf_node(&self, key: &[u8], mut offset: u64, buf: &mut [u8]) -> Result<u64> {
+impl<K: Key> BPTreeFileIndex<K> {
```

Comment:
`read_header_buf` in this file performs key comparison. It should be updated to use a method from `Key` trait

Reply:
Added key trait usage to `read_header_buf` (with vec copy for now)